modules/dnn/test/test_layers.cpp

```diff
@@ -618,6 +618,179 @@ TEST(Layer_LSTM_Test_Accuracy_with_, HiddenParams)
     normAssert(h_t_reference, outputs[0]);
 }
 
+typedef testing::TestWithParam<tuple<int, int>> Layer_GATHER_1d_Test;
+TEST_P(Layer_GATHER_1d_Test, Accuracy) {
+
+    int batch_size = get<0>(GetParam());
+    int axis = get<1>(GetParam());;
+
+    LayerParams lp;
+    lp.type = "Gather";
+    lp.name = "gatherLayer";
+    lp.set("axis", axis);
+    lp.set("real_ndims", 1);
+
+    Ptr<GatherLayer> layer = GatherLayer::create(lp);
+
+    std::vector<int> input_shape = {batch_size, 1};
+    std::vector<int> indices_shape = {batch_size, 1};
+    std::vector<int> output_shape = {batch_size, 1};
+
+    if (batch_size == 0){
+        input_shape.erase(input_shape.begin());
+        indices_shape.erase(indices_shape.begin());
+        output_shape.erase(output_shape.begin());
+    } else if (axis != 0) {
+        output_shape[1] = batch_size;
+    }
+
+    cv::Mat input = cv::Mat(input_shape, CV_32F, 1.0);
+    cv::Mat indices = cv::Mat(indices_shape, CV_32F, 0.0);
+    cv::Mat output_ref = cv::Mat(output_shape, CV_32F, 1.0);
```

Comment:
As we discussed offline the test us very weak. All indexes are zeros, all values are ones. I propose to use randShuffle for all indexes 0..size, randomize input values and prepare honest ground truth. See also  https://docs.opencv.org/4.x/d2/de8/group__core__array.html#ga6a789c8a5cb56c6dd62506179808f763